app/screens/invite/selection_invite_as.tsx

```diff
@@ -0,0 +1,179 @@
+            marginLeft: 4,
+        },
+        customMessage: {
+            marginHorizontal: 20,
+            marginTop: 8,
+        },
+        customMessageInput: {
+            height: 121,
+            backgroundColor: 'transparent',
+            ...typography('Body', 200, 'Regular'),
+            lineHeight: 20,
+            color: theme.centerChannelColor,
+            borderWidth: 1,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.16),
+            borderRadius: 4,
+            paddingTop: 12,
+            paddingBottom: 12,
+            paddingHorizontal: 16,
+        },
+        customMessageInputPlaceholder: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+        },
+    };
+});
+
+type SelectionInviteAsProps = {
+    guestEnabled: boolean;
+    selectedChannelsCount: number;
+    customMessageEnabled?: boolean;
+    customMessage?: string;
+    canChange?: boolean;
+    onGuestChange: (enabled: boolean) => void;
+    onSelectChannels: () => void;
+    onCustomMessageToggleChange: (enabled: boolean) => void;
+    onCustomMessageInputFocus: () => void;
+    onCustomMessageChange: (customMessage: string) => void;
+}
+
+export default function SelectionInviteAs({
+    guestEnabled,
+    selectedChannelsCount,
+    customMessageEnabled,
+    customMessage,
+    canChange = true,
+    onGuestChange,
+    onSelectChannels,
+    onCustomMessageToggleChange,
+    onCustomMessageInputFocus,
+    onCustomMessageChange,
+}: SelectionInviteAsProps) {
+    const {formatMessage} = useIntl();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const [isCustomMessageInputFocused, setIsCustomMessageInputFocused] = useState(false);
+
+    const handleOnCustomMessageInputFocus = () => {
+        setIsCustomMessageInputFocused(true);
+        onCustomMessageInputFocus();
+    };
+
+    const handleOnCustomMessageInputBlur = () => {
+        setIsCustomMessageInputFocused(false);
+    };
+
+    const customMessageInputStyle = useMemo(() => {
+        const style = [];
+
+        style.push(styles.customMessageInput);
+
+        if (isCustomMessageInputFocused) {
+            style.push({
+                borderWidth: 2,
+                borderColor: theme.buttonBg,
+            });
+        }
+
+        return style;
+    }, [isCustomMessageInputFocused, styles]);
+
+    return (
+        <View style={styles.container}>
+            <SettingSeparator/>
+            <SettingOption
+                type='toggle'
+                label={formatMessage({id: 'invite.invite_as.guest.label', defaultMessage: 'Invite as guest'})}
+                description={formatMessage({id: 'invite.invite_as.guest.description', defaultMessage: 'Guests are limited to selected channels'})}
+                action={onGuestChange}
+                selected={guestEnabled}
+                toggleItemProps={{disabled: !canChange}}
+                optionLabelTextStyle={{fontWeight: '600'}}
```

Comment:
Inline properties may (or may not) cause re-renders down the tree. Better extract them.